src/js/breadcrumb.ts

```diff
@@ -0,0 +1,53 @@
+import { Breadcrumb } from '@sentry/types';
+import { severityLevelFromString } from '@sentry/utils';
+
+type BreadcrumbCandidate = {
+  [K in keyof Partial<Breadcrumb>]: unknown;
+}
+
+/**
+ * Convert plain object to a valid Breadcrumb
+ */
+export function breadcrumbFromObject(candidate: BreadcrumbCandidate): Breadcrumb {
+  const breadcrumb: Breadcrumb = {};
+
+  if (typeof candidate.type === 'string') {
+    breadcrumb.type = candidate.type;
+  }
+  if (typeof candidate.level === 'string') {
+    breadcrumb.level = severityLevelFromString(candidate.level);
+  }
+  if (typeof candidate.event_id === 'string') {
+    breadcrumb.event_id = candidate.event_id;
+  }
+  if (typeof candidate.category === 'string') {
+    breadcrumb.category = candidate.category;
+  }
+  if (typeof candidate.message === 'string') {
+    breadcrumb.message = candidate.message;
+  }
+  if (typeof candidate.data === 'object' && candidate.data !== null) {
+    breadcrumb.data = candidate.data;
+  }
+  if (typeof candidate.timestamp === 'string') {
+    const timestamp = Date.parse(candidate.timestamp)
+    if (!isNaN(timestamp)) {
+      breadcrumb.timestamp = timestamp;
+    }
+  }
+
+  return breadcrumb;
+}
+
+/**
+ * Compares two breadcrumbs and returns true if they are equal
+ * Warning: This function does not check timestamp and level, data
```

Comment:
M: If the matter if to avoid duplication, all the fields would need to be considered.

Reply:
I excluded [the timestamp because it's not passed](https://github.com/getsentry/sentry-react-native/blob/a5d7a18015e3e11c7db6e69acd41e84c8d0c2b2b/ios/RNSentry.mm#L359) to the native layer, therefore it will be always different.
[The levels are also recalculated](https://github.com/getsentry/sentry-react-native/blob/a5d7a18015e3e11c7db6e69acd41e84c8d0c2b2b/src/js/wrapper.ts#L391) so they are not matching for the same breadcrumbs.
And data are being normalized so it's also tricky to check.

I suppose we will need to change how the breadcrumbs are passed to ios.

Reply:
Maybe[ this will solve it](https://github.com/getsentry/sentry-react-native/pull/2713#discussion_r1061360166).